src/ContractInteract/EIP20Gateway.js

```diff
@@ -535,8 +705,8 @@ class EIP20Gateway {
    */
   getLatestAnchorInfo() {
     return this.getAnchor().then((anchor) => {
-      anchor.getLatestStateRootBlockHeight().then((blockHeight) => {
-        anchor.getStateRoot(blockHeight).then((stateRoot) => {
+      return anchor.getLatestStateRootBlockHeight().then((blockHeight) => {
+        return anchor.getStateRoot(blockHeight).then((stateRoot) => {
```

Comment:
Isn't there an easier way to chain this? Like e.g.:
```js
return this.getAnchor()
  .then(anchor => anchor.getLatestStateRootBlockHeight())
  .then(blockHeight => anchor.getStateRoot(blockHeight))
  .then(stateRoot => ({
    blockHeight,
    stateRoot,
    ...
  }));
```

Note that my JS skills are very limited...

Reply:
I get 
`ReferenceError: anchor is not defined`
`ReferenceError: blockHeight is not defined`

Reply:
That's one of the prime use-cases for async/await:

```js
async getLatestAnchorInfo() {
  const anchor = await this.getAnchor();
  const blockHeight = await anchor.getLatestStateRootBlockHeight();
  const stateRoot = await anchor.getStateRoot(blockHeight);

  return {
    blockHeight,
    stateRoot,
  };
}